array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,11 @@
+/* exported zip */
+function zip(first, second) {
+  const newArray = [];
+  for (let i = 0; i < first.length; i++) {
```

Comment:
If you change up your condition expression you can always be sure to stop at the end of the shortest array.

```js
const end = Math.min(first.length, second.length);
for (let i = 0; i < end; i++) {
```